nixops/resources/ebs_volume.py

```diff
@@ -104,20 +113,24 @@ def create(self, defn, check, allow_reboot, allow_recreate):
 
         if self.state == self.MISSING:
 
-            if defn.size == 0 and defn.snapshot != "":
-                snapshots = self._conn.get_all_snapshots(snapshot_ids=[defn.snapshot])
-                assert len(snapshots) == 1
-                defn.size = snapshots[0].volume_size
-
-            if defn.snapshot:
-                self.log("creating EBS volume of {0} GiB from snapshot ‘{1}’...".format(defn.size, defn.snapshot))
+            if defn.volume_id != "":
+                vol_id = defn.volume_id
```

Comment:
what about the other volume attributes ? I think it'd be better to implement a _check method that pulls the volume attribute and persist to the state so in case it's an existent volume we run self._check() first